gwf/core.py

```diff
@@ -238,6 +230,41 @@ def target(self, name, inputs, outputs, **options):
         self._add_target(new_target)
         return new_target
 
+    def target_from_template(self, name, template, **options):
+        """Create a target and add it to the :class:`gwf.Workflow`.
+
+        This is syntactic sugar for creating a new :class:`~gwf.Target` and
+        adding it to the workflow. The target is also returned from the method
+        so that the user can directly manipulate it, if necessary. For example,
+        this allows assigning a spec to a target directly after defining it::
+
+            workflow = Workflow()
+            workflow.target_from_template('NewTarget', my_template())
+
+        This will create a new target named `NewTarget`, configure it based
+        on the specification in the template `my_template`, and
+        add it to the workflow.
+
+        :param str name: Name of the target.
+        :param tuple template: Target specification based on a template.
```

Comment:
Maybe change the description to "Target specification of the form (inputs, outputs, options, spec)"?